packages/data-stores/src/woocommerce-eligibility/reducer.ts

```diff
@@ -0,0 +1,44 @@
+import { combineReducers } from '@wordpress/data';
+import { TransferEligibility } from './types';
+import type { Action } from './actions';
+import type { Reducer } from 'redux';
+
+export const transferEligibility: Reducer<
+	{ [ key: number ]: TransferEligibility } | undefined,
+	Action
+> = ( state, action ) => {
+	switch ( action.type ) {
+		case 'TRANSFER_ELIGIBILITY_RECEIVE': {
+			return {
+				...state,
+				[ action.siteId ]: action.transferEligibility,
+			};
+		}
+	}
+
+	return state;
+};
+
+// Tracks product list fetching state
+export const isFetchingTransferEligibility: Reducer< boolean | undefined, Action > = (
+	state = false,
+	action
+) => {
+	switch ( action.type ) {
+		case 'TRANSFER_ELIGIBILITY_REQUEST':
+			return true;
```

Comment:
The "is fetching" state should be tracked by `siteId`. We can issue many parallel requests for different `siteId`'s that are completely independent.

Also, you can avoid it completely because `@wordpress/data` has built-in selectors like `isResolving` that track this automatically.